.github/workflows/ci.yml

```diff
@@ -0,0 +1,105 @@
+---
```

Comment:
Super tiny nitpick: Is there a need for this line at the top since this isn't a list?

Reply:
It's actually part of the YAML spec, but usually doesn't matter

Reply:
It's just a common convention with YAML files, as `---` is a document separator allowing multiple YAML documents within the same file. It's common to start YAML files with it, mostly I see it in Kubernetes manifests and GitHub Actions, and I have a habit of doing it.